### DISCUSSION TITLE
Detect the used SIMD instructions to compile Jolt.

### DISCUSSION THREAD

**Discussion by AndreaCatania** [jrouwe/JoltPhysics#585] [Q&A] [answered]
Hello, I'm building Jolt as static library using scons.

Jolt is being build with `SSE2 SSE4.1 SSE4.2 AVX AVX2 F16C LZCNT TZCNT` enabled; so, in order to produce a matching ABI I should define the following macros on the parent application too `USE_SSE2 USE_SSE4_1 USE_SSE4_2 USE_AVX USE_AVX2 USE_F16C USE_LZCNT USE_TZCNT`.

The Jolt cmake configuration is detecting the instruction set to use here https://github.com/jrouwe/JoltPhysics/blob/master/Jolt/Jolt.cmake#L525-L568 
So, I should create something like :point_up: in Scons to correctly define the macros.

I'm wondering if there is a smarter way though. Do you know an alternative way to know the active instruction set?

**Comment by jrouwe** [comment, marked as answer]
The SIMD instruction sets are not detected, you specify which ones you want to use by passing the variables to cmake, e.g.:

```
cmake [...] -DUSE_SSE4_2=Yes
```

The machine you compile the code on does not need to have the same capabilities as the machine you intend to distribute the compiled code to. So you choose what the minimum spec machine is and then define the variables that correspond to that minimum spec.

**Reply by AndreaCatania** [reply]
Ok, so there is no way to fetch these info from CMake (so that I can re-use the logic encoded in there); I need to re-create that logic in scons. Thx :call_me_hand:
